### PR TITLE
examples/mqttc_mbedtls: Add MQTT over TLS example

### PR DESCRIPTION
**Pull request by andreeaThePenguin**
## Summary
examples/mqttc_mbedtls: Add MQTT over TLS example using mbedTLS library

## Impact
N/A

## Testing
On ESP32 board

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



This PR appears to **marginally meet** the NuttX requirements. 

**Strengths:**

* **Summary** provides a clear description of the change.
* **Testing** section mentions the target platform.

**Areas for Improvement:**

* **Impact** section is entirely incomplete.  Even if no other areas are impacted, each should explicitly state "NO" with a brief justification (e.g., "Impact on build: NO - This change only affects example code.").  
* **Testing** should be more robust.  
    * Specify the build host details.
    * "Testing logs" are expected, even if they're simple demonstrations of functionality before and after the change. 

**Recommendation:**

Before submitting, revise the PR to thoroughly address the "Impact" section and provide more specific details in the "Testing" section.

**Comment by pkarashchenko** [comment]
I think I had similar changes stashed somewhere in local branch. But I extended existing example instead of creating a new one. Let me go thru the local branches this week and push my code as well, so we can compare.

File: netutils/mqttc/Kconfig [hunk 1]
```diff
@@ -25,12 +25,38 @@ config NETUTILS_MQTTC_EXAMPLE_STACKSIZE
...
+
+if NETUTILS_MQTTC_EXAMPLE_WITH_MBEDTLS
+
+config NETUTILS_MQTTC_EXAMPLE_STACKSIZE_WITH_MBEDTLS
+	int "Task's stack size"
+	default 8192
+
+endif
+
+config NETUTILS_MQTTC_WITH_MBEDTLS_VERSION
```
Comment (xiaoxiang781216) [inline netutils/mqttc/Kconfig:56]: why need two version config?

Comment (andreeaThePenguin) [reply netutils/mqttc/Kconfig:56]: I assumed a user may want to have different MQTT-C versions for each of the two example apps. This is a corner case of course, but may be useful.

Comment (xiaoxiang781216) [reply netutils/mqttc/Kconfig:56]: but the build doesn't support to download multiple package.

Comment (andreeaThePenguin) [reply netutils/mqttc/Kconfig:56]: Was not aware of this, so will remove the additional version config

**Comment by pkarashchenko** [comment]
Here is what I have https://github.com/apache/nuttx-apps/pull/2643

**Comment by andreeaThePenguin** [comment]
@pkarashchenko I checked your PR also - the extended example looks more neat, I think we can go with your app version

**Comment by xiaoxiang781216** [comment]
> @pkarashchenko I checked your PR also - the extended example looks more neat, I think we can go with your app version

so, let's close this pr and move the discussion to https://github.com/apache/nuttx-apps/pull/2643

**Pull request closed without merging**